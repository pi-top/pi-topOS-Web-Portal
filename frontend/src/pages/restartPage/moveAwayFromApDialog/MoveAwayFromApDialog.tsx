import React from "react";

import Dialog from "../../../components/atoms/dialog/Dialog";
import Button from "../../../components/atoms/button/Button";

import styles from "./MoveAwayFromApDialog.module.css";

export type Props = {
  active: boolean;
  piTopIpAddress: string;
  onSkip: () => void;
};

const getMessage = () => {
  return (
    <>
      <span className={styles.dialogTitle}>Reconnect to your Wi-Fi</span>
    </>
  );
};

export default ({
  active,
  piTopIpAddress,
  onSkip,
}: Props) => {

  const url = "http://" + piTopIpAddress + "/onboarding/reboot";

  return (
    <Dialog active={active} message={getMessage()} className={styles.dialog} dataTestId="move-away-from-ap-dialog">
      <div className={styles.content}>
        <div className={styles.message}>
          Your pi-top is not connected to the same network as your device.
          <br></br>
          <br></br>
          This probably means that your computer is connected to the 'pi-top' network and your pi-top was configured to connect to your Wi-Fi network in one of the previous steps.
          <br></br>
          <br></br>
          Please, make sure you connect your computer to your Wi-Fi network and refresh this page.
        </div>

        <div className={styles.buttons}>
          <div className={styles.refreshButtonContainer}>
            <Button className={styles.refreshButton} unstyled onClick={() => window.open(url, "_self")} >
              Refresh
            </Button>
          </div>

          <div className={styles.skipButtonContainer}>
            <Button className={styles.skipButton} unstyled onClick={() => onSkip()}>
              Skip
            </Button>
          </div>
        </div>

      </div>
    </Dialog>
  );
};
