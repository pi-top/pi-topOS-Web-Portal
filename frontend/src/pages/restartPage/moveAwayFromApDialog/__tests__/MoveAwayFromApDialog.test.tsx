import React, { ReactNode } from "react";
import ReactDom from "react-dom";
import {
  render,
  fireEvent,
  BoundFunction,
  GetByBoundAttribute,
  QueryByText,
  GetByText,
  RenderResult,
  QueryByBoundAttribute,
  wait,
} from "@testing-library/react";

import MoveAwayFromApDialog, { Props, getContent, getMessage } from "../MoveAwayFromApDialog";
import querySpinner from "../../../../../test/helpers/querySpinner";

const originalCreatePortal = ReactDom.createPortal;

window.open = jest.fn();

describe("MoveAwayFromApDialog", () => {
  let defaultProps: Props;
  let moveAwayFromApDialog: HTMLElement;
  let queryByText: BoundFunction<QueryByText>;
  let getByText: BoundFunction<GetByText>;
  let queryByTestId: BoundFunction<QueryByBoundAttribute>;
  let queryByAltText: BoundFunction<QueryByBoundAttribute>;
  let queryByLabelText: BoundFunction<QueryByBoundAttribute>;
  let getByLabelText: BoundFunction<GetByBoundAttribute>;
  let rerender: RenderResult["rerender"];

  const ipAddress = "1.1.1.1";

  beforeEach(() => {
    defaultProps = {
      active: true,
      piTopIpAddress: ipAddress,
      onSkip: jest.fn(),
    };

    ReactDom.createPortal = jest.fn();
    const createPortalMock = ReactDom.createPortal as jest.Mock;
    createPortalMock.mockImplementation((element: ReactNode) => element)(
      ({
        container: moveAwayFromApDialog,
        queryByText,
        queryByTestId,
        queryByAltText,
        queryByLabelText,
        getByLabelText,
        getByText,
        rerender,
      } = render(<MoveAwayFromApDialog {...defaultProps} />))
    );
  });
  afterEach(() => {
    ReactDom.createPortal = originalCreatePortal;
  });

  it("renders the correct message", () => {
    const content = getMessage();
    expect(content).toMatchSnapshot();
  });

  it("renders the correct content", () => {
    const content = getContent();
    expect(content).toMatchSnapshot();
  });

  it("renders Refresh button", () => {
    expect(queryByText("Refresh")).toBeInTheDocument();
  });

  it("renders Skip button", () => {
    expect(queryByText("Skip")).toBeInTheDocument();
  });

  it("onSkip is called when Skip button is pressed", () => {
    fireEvent.click(getByText("Skip"));
    expect(defaultProps.onSkip).toHaveBeenCalled();
  });

  it("redirects to IP when Refresh button is pressed", async () => {
    fireEvent.click(getByText("Refresh"));
    await wait();
    expect(window.open).toHaveBeenNthCalledWith(
      1,
      "http://" + ipAddress + "/onboarding/reboot",
      "_self"
    )
  });

});
