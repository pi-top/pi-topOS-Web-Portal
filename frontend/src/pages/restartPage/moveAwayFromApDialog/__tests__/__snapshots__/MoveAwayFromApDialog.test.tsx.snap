// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoveAwayFromApDialog renders the correct content 1`] = `
<React.Fragment>
  Your pi-top is not connected to the same network as your device.
  <br />
  <br />
  This probably means that your computer is connected to the 'pi-top' network and your pi-top was configured to connect to your Wi-Fi network in one of the previous steps.
  <br />
  <br />
  Please, make sure you connect your computer to your Wi-Fi network and refresh this page.
</React.Fragment>
`;

exports[`MoveAwayFromApDialog renders the correct message 1`] = `
<React.Fragment>
  <span
    className="dialogTitle"
  >
    Reconnect to your Wi-Fi
  </span>
</React.Fragment>
`;
